plenum/test/node_catchup/test_incorrect_catchup_request.py

```diff
@@ -0,0 +1,99 @@
+
+    req = CatchupReq(leger_id, 0, end, catchup_till)
+    _process_catchup_req(req,
+                         _check_discard,
+                         looper,
+                         txnPoolNodeSet,
+                         sdk_pool_handle,
+                         sdk_wallet_client,
+                         monkeypatch)
+
+
+def test_receive_incorrect_catchup_request_with_start_greater_end(looper,
+                                                                  txnPoolNodeSet,
+                                                                  sdk_pool_handle,
+                                                                  sdk_wallet_client,
+                                                                  monkeypatch):
+    start = 10
+    end = 5
+
+    def _check_discard(msg, reason, logMethod=logging.error, cliOutput=False):
+        assert reason.find("not able to service since "
+                           "start = {} greater than "
+                           "end = {}"
+                           .format(start, end))
+
+    req = CatchupReq(leger_id, start, end, 11)
+    _process_catchup_req(req,
+                         _check_discard,
+                         looper,
+                         txnPoolNodeSet,
+                         sdk_pool_handle,
+                         sdk_wallet_client,
+                         monkeypatch)
+
+
+def test_receive_incorrect_catchup_request_with_catchuptill_greater_ledger_size(
+        looper,
+        txnPoolNodeSet,
+        sdk_pool_handle,
+        sdk_wallet_client,
+        monkeypatch):
+    catchup_till = 100
+    req = CatchupReq(leger_id, 0, 10, catchup_till)
+    ledger_size = txnPoolNodeSet[0].ledgerManager.getLedgerForMsg(req).size
+
+    def _check_discard(msg, reason, logMethod=logging.error, cliOutput=False):
+        assert reason.find("not able to service since "
+                           "catchupTill = {} greater than "
+                           "ledger size = {}"
+                           .format(catchup_till, ledger_size))
+
+    _process_catchup_req(req,
+                         _check_discard,
+                         looper,
+                         txnPoolNodeSet,
+                         sdk_pool_handle,
+                         sdk_wallet_client,
+                         monkeypatch)
+
+
+def _process_catchup_req(req: CatchupReq,
+                         check_discard,
+                         looper,
+                         txnPoolNodeSet,
+                         sdk_pool_handle,
+                         sdk_wallet_client,
+                         monkeypatch):
+    sdk_send_random_and_check(looper,
+                              txnPoolNodeSet,
+                              sdk_pool_handle,
+                              sdk_wallet_client,
+                              4)
+    ledger_manager = txnPoolNodeSet[0].ledgerManager
+    monkeypatch.setattr(ledger_manager.owner, 'discard', check_discard)
```

Comment:
A test will pass if `check_discard` is not called at all. The tests must ensure that `node.discard` is called.

Reply:
If this messages will not discard, merkle tree will have exception and test will fail.